YakimS/assignment2/BinaryTree.java

```diff
@@ -0,0 +1,231 @@
+package CodeUAss1Package;
+
+import javax.management.openmbean.KeyAlreadyExistsException;
+import java.util.*;
+
+/**
+ * This class contains the root to a binary tree that can have only
+ * one node for each possible key.
+ * @param <T> is the type of data stored in the tree nodes value
+ */
+public class BinaryTree<T> {
+
+    private Node root;
+    private Set<Integer> keySet;

```

Comment:
The whole tree has generic parameter `<T>`, so `keySet` should probably also use it.

Reply:
The generic parameter is referred to the type of data the tree stores in his nodes, and I choose to store the key as integers. Can you please explain why should I use the keys of the nodes to be the same type as the value they reference?

Reply:
You've answered my question at the next change request.